### PR TITLE
Jetpack App (Basics): Include atomic sites

### LINKED ISSUE #14619: Jetpack App (Basics): Include Atomic Sites
Parent #14293 

- [x] Include Atomic sites in the API call
- [x] Validate Login flow continues to work correctly for No Jetpack Sites & Not a Jetpack Site

### PR DESCRIPTION
**Pull request by ashiagr**
Fixes #14619

This PR includes atomic sites for the Jetpack app.

To test:

1. Install and launch Jetpack app.
2. Login with an account with an atomic site.
3. Notice that login is successful and the site is visible in the site picker.

## Regression Notes
1. Potential unintended areas of impact 🟢 
2. What I did to test those areas of impact (or what existing automated tests I relied on) 🟢 
3. What automated tests I added (or what prevented me from doing so) 🟢 

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java [hunk 1]
```diff
@@ -384,9 +384,9 @@ public static boolean isScanEnabled(boolean scanPurchased, SiteModel site) {
     }
 
     @NonNull
-    public static FetchSitesPayload getFetchSitesPayload() {
+    public static FetchSitesPayload getFetchSitesPayload(boolean isJetpackApp) {
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:387]: 👍  on this change ❤️

**Review by zwarm — APPROVED** [review]
All looks good @ashiagr . I can log in using an atomic site. 
I'll approve this, but won't merge because I don't know how you are managing the Login library updates.

**Comment by ashiagr** [comment]
Thank @zwarm, we're still deciding whether we should include atomic sites for the Jetpack app. Till that time let's keep this PR open. The login library will be updated accordingly.

**Comment by ashiagr** [comment]
Closing this PR as these changes are not required and we'd keep the behaviour similar to http://cloud.jetpack.com/ that includes only the Jetpack sites (no atomic sites).

cc @leandroalonso

**Pull request closed without merging**